### PR TITLE
add zerotier basic testcases

### PR DESCRIPTION
**Pull request by ahmedelsayed-93**

### CODE REVIEW

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_storagepools_apis.py [hunk 1]
```diff
@@ -90,27 +93,29 @@ def test003_post_storagepool(self):
         levels = ['raid0', 'raid1', 'raid5', 'raid6', 'raid10', 'dup', 'single']
         metadata = self.random_item(levels)
         data = self.random_item(levels)
-        body = {"name":name, "metadataProfile":metadata, "dataProfile":data, "devices":[]}
-        response = self.storagepool_api.post_storagepools(nodeid, name)
+        body = {"name":name, "metadataProfile":metadata, "dataProfile":data, "devices":['/dev/sda']}
+        response = self.storagepool_api.post_storagepools(nodeid, body)
         self.assertEqual(response.status_code, 201)
-        mountpoint = response.header['location']
+        mountpoint = response.headers['Location']
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_storagepools_apis.py]: any data should be returned in response body not in the header

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_vms_apis.py [hunk 2]
```diff
@@ -16,8 +17,8 @@ def setUp(self):
 
         self.lg.info('Create virtual machine (VM0) on node (N0)')
         self.vm_id = self.random_string()
-        self.vm_mem = randint(1, 4)*1024
-        self.vm_cpu = randint(1, 4)
+        self.vm_mem = 1024
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_vms_apis.py]: should it be a specific number?

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_vms_apis.py [hunk 3]
```diff
@@ -16,8 +17,8 @@ def setUp(self):
...
+        self.vm_cpu = 1
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_vms_apis.py]: should it be a specific number?

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_vms_apis.py [hunk 4]
```diff
@@ -130,15 +134,15 @@ def test004_put_nodes_vms_vmid(self):
         #. Update virtual machine with missing parameters, should fail with 400.
         """
         self.lg.info('Create virtual machine (VM0) on node (N0)')
-        vm_id = self.random_string()
-        vm_mem = randint(1, 4)*1024
-        vm_cpu = randint(1, 4)
+        vm_id = self.vm_id
+        vm_mem = 2*1024
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_vms_apis.py]: should it be a specific number?

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_vms_apis.py [hunk 5]
```diff
@@ -130,15 +134,15 @@ def test004_put_nodes_vms_vmid(self):
...
+        vm_cpu = randint(1, 2)
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_vms_apis.py]: only two values?

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_bridges_apis.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        response = self.bridges_api.get_nodes_bridges_bridgeid(self.nodeid, self.name)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(self.name, response.json()['name'])
+        self.assertEqual(self.networkMode, response.json()['settings'])
+        self.assertEqual('up', response.json()['status'])
+
+        self.lg.info('Get nonexisting bridge, should fail with 404')
+        response = self.bridges_api.get_nodes_bridges_bridgeid(self.nodeid, 'fake_bridge')
+        self.assertEqual(response.status_code, 404)
+
```
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_bridges_apis.py]: compare result with python client

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_bridges_apis.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        #. Get random node (N0).
+        #. Create bridge (B0) on node (N0).
+        #. List node (N0) bridges, should succeed with 200.
+        """
+        self.lg.info('Get bridge (B0), should succeed with 200')
+        response = self.bridges_api.get_nodes_bridges(self.nodeid)
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(self.name, [x['name'] for x in response.json()])
+
```
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_bridges_apis.py]: compare result with client bridge list

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_bridges_apis.py [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+        body = {"name":name,
+                "hwaddr":hwaddr,
+                "networkMode":networkMode,
+                "nat":nat,
+                "setting":settings[networkMode]}
+
+        response = self.bridges_api.post_nodes_bridges(self.nodeid, body)
+        self.assertEqual(response.status_code, 201, response.content)
+
```
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_bridges_apis.py]: make sure that the bridge created with client

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_storageclusters_apis.py [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+        #. Deploy new storage cluster (SC0)
+        #. Get storage cluster (SC0), should succeed with 200
+        #. Get nonexisting storage cluster (SC0), should fail with 404
+        """
+        self.lg.info('Get storage cluster (SC0), should succeed with 200')
+        response = self.storageclusters_api.get_storageclusters_label(self.label)
+        self.assertEqual(response.status_code, 200)
+        for key in ['label', 'driveType', 'nodes']:
+            self.assertEqual(response.json()[key], self.body[key])
+
```
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_storageclusters_apis.py]: test status of created storagecluster , should be empty, deploying, ready or error ?

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_storagepools_apis.py [hunk 10]
```diff
@@ -384,6 +420,7 @@ def test015_post_storagepool_filesystem_snapshot(self):
         response = lf.storagepool_api.post_filesystems_snapshots(self.nodeid, self.storagepool_name, self.fs_name, body)
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_storagepools_apis.py]: self not lf

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_storagepools_apis.py [hunk 11]
```diff
@@ -1,8 +1,8 @@
-from random import randint
+import  random
 from api_testing.testcases.testcases_base import TestcasesBase
 from api_testing.grid_apis.apis.storagepools_apis import StoragepoolsAPI
-
-
+from api_testing.python_client.client import Client
+import unittest, time
 
 class TestStoragepoolsAPI(TestcasesBase):
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_storagepools_apis.py]: please add verifying methods using the client.btrfs as every storagepool API is implemented using this it.
to knw which btrfs argument u need to knw please check the following link:
https://github.com/Jumpscale/jumpscale_core8/blob/8.2.0/lib/JumpScale/sal/g8os/StoragePool.py

**Pull request merged**